core/pkg/server/handler.go

```diff
@@ -1244,6 +1228,32 @@ func (h *Handler) flushPartialHistory(useStep bool, nextStep int64) {
 	h.handleHistoryDirectly(historyRecord)
 }
 
+// updateSummary updates the summary based on the current history step.
+func (h *Handler) updateSummary() {
+	updates, err := h.runSummary.UpdateSummaries(h.partialHistory)
+
+	// We continue despite errors to update as much of the summary as we can.
+	if err != nil {
+		h.logger.CaptureError(
+			fmt.Errorf("handler: error updating summary: %v", err))
+	}
+
+	if len(updates) == 0 {
+		return
+	}
+
+	// We must forward these changes to the Sender which uses them to build
+	// its own summary.
+	h.fwdRecord(&service.Record{
+		RecordType: &service.Record_Summary{
+			Summary: &service.SummaryRecord{
+				Update: updates,
+			},
+		},
+		Control: &service.Control{Local: true},
```

Comment:
wait again same comment as with the time update... local means it won't be added to the transaction log

Reply:
hmm... I think you have a point here because we don't save partial history records to the transaction log, and this codepath is only used for partial history records.

Reply:
but we convert them to full histories and these are stored in the transaction log, so maybe it should move to the full history code path? or some way moving from partial history to full history once it is committed

Reply:
ah i see what you meant

Reply:
i updated this so that records created on this codepath are written to the transaction log